### PR TITLE
RPE-508: Add spring session

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Incorporate spring session](https://tools.hmcts.net/jira/browse/RPE-508)

### Change description ###

Simple configuration to manage session with jdbc for multiple app instances

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/feature/webconsole/AdminAccessTest.java [hunk 1]
```diff
@@ -19,32 +21,39 @@
     @Test
     public void should_verify_login_logout_journey() {
         RequestSpecification specification = requestSpecification();
-        String sessionCookieName = "JSESSIONID";
+        String javaSessionCookieName = "JSESSIONID";
+        String sessionCookieName = "SESSION";
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/feature/webconsole/AdminAccessTest.java]: are 2 session cookies needed??

File: src/functionalTest/java/uk/gov/hmcts/reform/feature/webconsole/AdminAccessTest.java [hunk 2]
```diff
@@ -19,32 +21,39 @@
...
             .formParam("password", testAdminPassword)
             .post("/login")
             .then()
             .statusCode(FOUND.value())
             .extract()
-            .cookie(sessionCookieName);
+            .cookies();
+
+        String javaSessionCookieValue = cookies.getOrDefault(javaSessionCookieName, "");
+        String sessionCookieValue = cookies.getOrDefault(sessionCookieName, "");
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/feature/webconsole/AdminAccessTest.java]: didn't something go wrong if the cookie is not there? ie, test should fail here

File: src/main/resources/db/migration/V003__Spring_session_schema.sql [hunk 3]
```diff
@@ -0,0 +1,20 @@
+CREATE TABLE SPRING_SESSION (
+	SESSION_ID CHAR(36) NOT NULL,
+	CREATION_TIME BIGINT NOT NULL,
+	LAST_ACCESS_TIME BIGINT NOT NULL,
+	MAX_INACTIVE_INTERVAL INT NOT NULL,
+	PRINCIPAL_NAME VARCHAR(100),
```
Comment (rkondratowicz) [inline src/main/resources/db/migration/V003__Spring_session_schema.sql]: Use spaces, not tabs.
Also `cmd+alt+l` to format.

Comment (doncem) [reply src/main/resources/db/migration/V003__Spring_session_schema.sql]: flipping java people.

File: src/main/resources/application.yaml [hunk 4]
```diff
@@ -28,6 +31,8 @@ spring:
       poolName: FeatureToggleHikariCP
       maxLifetime: 7200000
       connectionTimeout: 30000
+  session:
+    store-type: jdbc
```
Comment (rkondratowicz) [inline src/main/resources/application.yaml]: timeout should be configure here I think

**Pull request merged**